### PR TITLE
Game engine

### PR DESCRIPTION
**Pull request by Flaagrah**
Created basic framework for the game and created unit tests for Game and Board. All unit tests currently pass.

### CODE REVIEW

**Review by pavel6767 — APPROVED** [review]
this is amazing!
I ran tests locally and not sure if this is an issue, but seeing a statement printed twice whenever a team wins
i.e. Red Won
Red Won

File: server/engine/Board.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	constructor() {
+		this.wordList = new Array(25);
+		var i = 0;
+		var dictSize = Dictionary.length;
+		var wordMap = {};
+		var red = 9;
+		var blue = 8;
+		var civilian = 7;
+		var assassin = 1;
+		for (i = 0;i<25;i++)
```
Comment (aeciorc) [inline server/engine/Board.js]:
```suggestion
		this.wordList = new Array(25);
		const dictSize = Dictionary.length;
		const wordMap = {};
		let red = 9;
		let blue = 8;
		let civilian = 7;
		let assassin = 1;
		for (let i = 0;i<25;i++)

```

File: server/engine/Board.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+						this.state = WordStates.CIVILIAN;
+						civilian--;
+					}
+				} else {
+					this.state = WordStates.BLUE;
+					blue--;
+				}
+			} else {
+				red--;
+			}
```
Comment (aeciorc) [inline server/engine/Board.js]: The way you assign the words is pretty clever 👍

File: server/engine/GameWord.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const word_state = require("./WordStates.js");
```
Comment (aeciorc) [inline server/engine/GameWord.js]: use camelCase

File: server/engine/GameWord.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+		return this.val;
+	}
+
+	get_person() {
+		return this.person;
+	}
+
+	get_chosen() {
+		return this.chosen;
+	}
```
Comment (aeciorc) [inline server/engine/GameWord.js]: Here as well

File: server/engine/GameWord.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+	choose() {
+		if (this.chosen = true) {
+			return false;
+		}
+		chosen = true;
+		return true;
+	}
+}
```
Comment (aeciorc) [inline server/engine/GameWord.js]:
```suggestion
		if (this.chosen) {
			return false;
		}
		this.chosen = true;
		return true;
	}
}

```

File: server/engine/WordStates.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+var word_state = {
```
Comment (aeciorc) [inline server/engine/WordStates.js]: camelCase

**Review by aeciorc — CHANGES_REQUESTED** [review]
Other things to do merging this PR:
- Move the stuff from app.js to another script that we can use for testing
- Favour const over let and let over var

**Review by aeciorc — APPROVED** [review]

**Pull request merged**